src/Pages/ChangePassword.cshtml.cs

```diff
@@ -65,99 +66,101 @@ public async Task<IActionResult> OnGet()
                 return RedirectToPage("/index");
             }
 
-            await SetCurrency();
+            await SetCurrencyList();
+            await SetUserCurrency();
             return Page();
         }
 
-        private async Task SetCurrency()
+        private async Task SetUserCurrency()
         {
             var userId = int.Parse(User.FindFirst(ClaimTypes.Name).Value);
-            UserModel = await UserDatabase.GetUserAsync(userId);
-            CurrencyOptions = new List<string>();
-            if (Cache.TryGetValue(CacheKeys.CurrencyList, out List<string> cachedCurrencyList))
+            var (currency, exception) = await CurrencyConverter.GetUserCurrency(userId);
+            if (exception != null)
             {
-                CurrencyOptions.AddRange(cachedCurrencyList);
-                //Check if user's currency is cached
-                if (Cache.TryGetValue(CacheKeys.UserCurrency, out Currency cachedCurrency))
+                switch (exception)
                 {
-                    if (UserModel.Currency.Equals(cachedCurrency.Code)) return;
+                    case HttpListenerException ex:
+                        _logger.LogInformation(ex.ToString());
+                        ErrorMessage = "Failed to load currency information!";
+                        break;
+                    case HttpRequestException ex:
+                        _logger.LogInformation(ex.ToString());
+                        ErrorMessage = "Failed to load currency information!";
+                        break;
+                    default:
+                        // Failed to even get the user currency from the database, the big one
+                        _logger.LogInformation(exception.ToString());
+                        ErrorMessage = "Failed to connect to database!";
+                        break;
                 }
```

Comment:
I think property `FailedToGetCurrencyList` should be set to true instead of `ErrorMessage`?
Maybe I myself missed this, just noting it to remember

Reply:
The property FailedToGetCurrencyList  is not used anywhere in the code though